docs/features/readonly-variables-and-parameters.md

```diff
@@ -0,0 +1,61 @@
+### High Level Language Proposal:  
+Allow 'readonly' to be used on local variables and on parameters. 
+
```

Comment:
I do not think "inserted into use sites"  is required or observable for local constants. Decimal locals have 3 different possible implementations (default, local, field) and compiler just choses the cheapest for a given value - user cannot see the difference.

Conversely, regular locals can optimize into embedded literal values if data flow is simple enough.

The readonly locals do look a lot like consts with more types allowed and with initializers that can cause or observe sideeffects. 

I think generalizing const could be a nicer approach since it would not be introducing an overlapping concept and const is also a shorter keyword.